segmentstore/server/src/main/java/io/pravega/segmentstore/server/WriterSegmentProcessor.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines a general Segment Processor for the Storage Writer.
+ */
+public interface WriterSegmentProcessor extends AutoCloseable {
```

Comment:
This interface has no call defined indicating processing. Given that it is the contract of a `Processor`, shouldn't it have a method that indicates processing?

Reply:
It has two methods:
1. `add(Operation)`
2. `flush()`

Together, they perform some sort of processing.